File: p4th/HeejiYun/backend/src/main/java/com/example/backend/entity/board/Board.java
package com.example.backend.entity.board;

import com.example.backend.entity.member.Member;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;

@Data
@Entity
@NoArgsConstructor
public class Board {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long boardNo;

    @Column(length = 128, nullable = false)
    private String title;

    @Column(length = 32, nullable = false)
    private String writer;

    @Column
    private String content;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "board_category_id")
    private BoardCategory boardCategory;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "member_id")
    private Member member;

    @CreationTimestamp
    private Date regDate;

    @UpdateTimestamp
    private Date updDate;

    public Board(String title, String writer, String content) {
        this.title = title;
        this.writer = writer;
        this.content = content;
    }
}

File: p4th/HeejiYun/frontend/lib/api/board.dart
class Board {
  int boardNo;
  String title;
  String writer;
  String content;
  String categoryName;
  String regDate;

  Board({
    required this.boardNo,
    required this.title,
    required this.writer,
    required this.content,
    required this.categoryName,
    required this.regDate
  });

  factory Board.fromJson(Map<String, dynamic> json) {
    return Board(
      boardNo: json["boardNo"] as int,
      title: json["title"] as String,
      writer: json["writer"] as String,
      content: json["content"] as String,
      categoryName: json["categoryName"] as String,
      regDate: json["regDate"] as String,
    );
  }
}